gitpod-io/gitpod
# [JetBrains EAP] Add actions `Copy URL` and `Copy Web URL` to the terminal's ports context menu

## Description

This PR enables the Port Forwarding Action initially implemented by @andreafalzetti.

When the user expands the "Forwared Port context menu" there's an option for copying the URL from that port:
<img width="575" alt="image" src="https://user-images.githubusercontent.com/418083/199527362-bfd75ddf-fdb6-4e38-bf3b-2d7ffad3d613.png">

- `Copy URL` copies the forwarded URL (e.g. `http://127.0.0.1:8000`)
- `Copy Web URL` copies the exposed URL (e.g. `https://8000-gitpodio-gitpod-37qofs346l3.ws-eu73.gitpod.io`)

The 'Copy Web URL' is also available in the non-forwared port context menu:
<img width="566" alt="image" src="https://user-images.githubusercontent.com/418083/199527660-6fe627fb-046f-42e5-a00e-ecfd75781814.png">

Note: The action is disabled when used from the Search Menu, and will only be displayed (greyed-out) if the user checks the "Include non-project items" box.

<img width="686" alt="image" src="https://user-images.githubusercontent.com/418083/199546248-95857fc0-bbf3-40df-9a91-38337668d6ac.png">

## Related Issue(s)

Initially implemented in https://github.com/gitpod-io/gitpod/pull/13990

## How to test

1. In the Preview Environment of this PR, select **IntelliJ EAP** as your editor
2. Start a workspace from https://github.com/gitpod-io/empty
3. Open a terminal and run `python -m http.server`
4. Observe that the port is automatically forwarded
5. Click the forwarded port context menu and select "Copy URL" and confirm if the URL has been correctly copied to the clipboard
6. Click the forwarded port context menu and select "Copy Web URL" and confirm if the URL has been correctly copied to the clipboard
7. Click the non-forwarded port context menu and select "Copy Web URL" and confirm if the URL has been correctly copied to the clipboard

## Release Notes
```release-note
In JetBrains EAP IDEs, users now have the option to copy the URL from the terminal's ports context menu.
```

## Werft options:
